src/components/Login/utils.test.tsx

```diff
@@ -2,28 +2,55 @@ import { decodeQr } from "./utils";
 
 describe("decodeQr", () => {
   it("should fail for old, deprecated QR codes", () => {
-    expect.assertions(1);
+    expect.assertions(2);
     const code = "1e4457bc-f7d0-4329-a344-f0e3c75d8dd4";
     expect(() => decodeQr(code)).toThrow("Invalid format");
+    const url = "https://example.com";
+    expect(() => decodeQr(url)).toThrow("Invalid format");
   });
 
   it("should decode the correct qr code and return endpoint and key", () => {
-    expect.assertions(2);
+    expect.assertions(4);
     const code = `{"key": "1e4457bc-f7d0-4329-a344-f0e3c75d8dd4","endpoint": "https://somewhere.com"}`;
-    const { endpoint, key } = decodeQr(code);
-    expect(endpoint).toBe("https://somewhere.com");
-    expect(key).toBe("1e4457bc-f7d0-4329-a344-f0e3c75d8dd4");
+    const url =
+      "https://somewhere.com?key=1e4457bc-f7d0-4329-a344-f0e3c75d8dd4&endpoint=https://somewhere.com";
```

Comment:
let's change the base URL or endpoint here so that it's clearer which one is the endpoint

Reply:
```suggestion
      "https://this-is-the-base-url.com?key=1e4457bc-f7d0-4329-a344-f0e3c75d8dd4&endpoint=https://endpoint.com";
```